crates/oxc_linter/src/rules/react/jsx_no_script_url.rs

```diff
@@ -0,0 +1,274 @@
+                        };
+                        if prop_value.as_string_literal().is_some_and(|val| {
+                            is_link_attribute(
+                                component_name.as_str(),
+                                attr.name.get_identifier().name.to_string(),
+                                ctx,
+                            ) && JS_SCRIPT_REGEX.captures(&val.value).is_some()
+                        }) {
+                            ctx.diagnostic(jsx_no_script_url_diagnostic(attr.span()));
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    fn from_configuration(value: Value) -> Self {
+        let mut components: FxHashMap<String, Vec<String>> = FxHashMap::default();
+        if let Some(arr) = value.get(0).and_then(Value::as_array) {
+            for component in arr {
+                let name = component.get("name").and_then(Value::as_str).unwrap_or("").to_string();
+                let props =
+                    component.get("props").and_then(Value::as_array).map_or(vec![], |array| {
+                        array
+                            .iter()
+                            .map(|prop| prop.as_str().map_or(String::new(), String::from))
+                            .collect::<Vec<String>>()
+                    });
+                components.insert(name, props);
+            }
+            Self(Box::new(JsxNoScriptUrlConfig {
+                include_from_settings: value.get(1).is_some_and(|conf| {
+                    conf.get("includeFromSettings").and_then(Value::as_bool).is_some_and(|v| v)
+                }),
+                components,
+            }))
+        } else {
+            Self(Box::new(JsxNoScriptUrlConfig {
+                include_from_settings: value.get(0).is_some_and(|conf| {
+                    conf.get("includeFromSettings").and_then(Value::as_bool).is_some_and(|v| v)
+                }),
+                components: FxHashMap::default(),
+            }))
+        }
+    }
+
+    fn should_run(&self, ctx: &ContextHost) -> bool {
+        ctx.source_type().is_jsx()
+    }
+}
+
+#[test]
+fn test() {
+    use crate::tester::Tester;
+
+    let pass = vec![
+        (r#"<a href="https://reactjs.org"></a>"#, None, None),
+        (r#"<a href="mailto:foo@bar.com"></a>"#, None, None),
+        (r##"<a href="#"></a>"##, None, None),
+        (r#"<a href=""></a>"#, None, None),
+        (r#"<a name="foo"></a>"#, None, None),
+        (r#"<a href={"javascript:"}></a>"#, None, None),
+        (r#"<Foo href="javascript:"></Foo>"#, None, None),
+        ("<a href />", None, None),
+        (
+            r#"<Foo href="javascript:"></Foo>"#,
+            None,
+            Some(
+                serde_json::json!({ "settings": {"react": {"linkComponents": [{ "name": "Foo", "linkAttribute": ["to", "href"] }]} } }),
+            ),
+        ),
+        (
+            r#"<Foo other="javascript:"></Foo>"#,
```

Comment:
FYI, I added an extra test to `pass`,  as it checks for the link attributes not to have the `javascript:` value.(It seems that the original implementation of the rule does that, it just doesn't have a test case for it). I can remove it though if deemed as not necessary.